src/Feature.ts

```diff
@@ -0,0 +1,69 @@
+export type FeatureFlag = {
+    [internalName: string]: boolean;
+};
+
+/**
+ * @todo documentation
+ *
+ * @since {date}
+ */
+export class Feature {
+    static readonly TASK_STATUS_MENU = new Feature(
+        'TASK_STATUS_MENU',
+        0,
+        'Enables a right click menu for each task to allow you to select the task Status from the available next transition states.',
+        'Task Status Menu',
+        false,
+        false,
+    );
+
+    static get values(): Feature[] {
```

Comment:
How would this be used in practice?

Something simple like `if (Feature.enabled(Feature.TASK_STATUS_MENU))` would be ideal...

Reply:
Later: interaction with features is via the settings class.